proto/allocation/allocation.proto

```diff
@@ -87,6 +87,24 @@ message MetaPatch {
 
 // LabelSelector used for finding a GameServer with matching labels.
 message LabelSelector {
+  // Labels to match.
+  map<string, string> matchLabels = 1;
+}
+
+// GameServerSelector used for finding a GameServer with matching filters.
+message GameServerSelector {
   // Labels to match.
 	map<string, string> matchLabels = 1;
-}
\ No newline at end of file
+  enum GameServerState {
+     READY = 0;
+     ALLOCATED = 1;
+  };
+  GameServerState gameServerState = 2;
+  PlayerSelector players = 3;
+}
+
+// PlayerSelector is filter for player capacity values.
```

Comment:
Consider adding a comment that minAvailable must be less than maxAvailable.

Reply:
I like it!